core/src/wallet/cosmos/CosmosLikeAccount.cpp

```diff
@@ -600,20 +610,38 @@ namespace ledger {
                         auto buildFunction = [self](const CosmosLikeTransactionBuildRequest &request) {
                                 auto currency = self->getWallet()->getCurrency();
                                 auto tx = std::make_shared<CosmosLikeTransactionApi>();
-                                tx->setAccountNumber(self->getAccountUid());
+                                tx->setAccountNumber(self->getAccountNumber());
                                 tx->setCurrency(self->getWallet()->getCurrency());
                                 tx->setFee(request.fee);
                                 tx->setGas(request.gas);
                                 tx->setMemo(request.memo);
                                 tx->setMessages(request.messages);
-                                tx->setSequence(request.sequence);
```

Comment:
`sequence` can maybe be removed from `CosmosLikeTransactionBuildRequest`

Reply:
I'll make it optional instead then. If the live wants to set the sequence themselves we'll let them (in case they want to build multiple transactions "offline" before broadcasting, or keeping an internal sequence count or something)

Reply:
So probably `request.sequence.empty() ? self->getSequence() : request.sequence`